integrations/venice-cuda/src/main/java/com/linkedin/venice/cuda/GPUBackedDaVinciClient.java

```diff
@@ -0,0 +1,138 @@
+package com.linkedin.venice.cuda;
+
+import com.linkedin.davinci.client.DaVinciClient;
+import com.linkedin.davinci.client.DaVinciConfig;
+import com.linkedin.davinci.client.DaVinciRecordTransformerConfig;
+import com.linkedin.davinci.client.DaVinciRecordTransformerFunctionalInterface;
+import com.linkedin.davinci.client.factory.DaVinciClientFactory;
+import java.io.Closeable;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * A wrapper around DaVinciClient that manages GPU-backed storage for Venice data.
+ * This class handles the lifecycle of GPU transformers across different versions.
+ * 
+ * @param <K> Key type
+ * @param <V> Value type
+ */
+public class GPUBackedDaVinciClient<K, V> implements Closeable {
+  private static final Logger LOGGER = LogManager.getLogger(GPUBackedDaVinciClient.class);
+
+  private final String storeName;
+  private final DaVinciClientFactory daVinciClientFactory;
+  private DaVinciClient<K, V> daVinciClient;
+  // TODO: Think through what parameterization makes sense for the transformer relative to this class
+  private final ConcurrentSkipListMap<Integer, GPUDaVinciRecordTransformer<K, V, V>> versionTransformers;
+  private final int gpuDeviceId;
+  private final int initialCapacity;
+
+  /**
+   * Constructor with custom transformer factory
```

Comment:
This doc is out of date (originally I was having this take the factory as a parameter, but decided that was dumb).  Need to update.